### ISSUE TITLE
Яндекс.Практикум:Код-ревью

### ISSUE THREAD

**Issue by zlocate** [romanRysev/romanRysev.github.io#2]
Здравствуйте. Благодарю за проделанную работу.
К сожалению текущая реализация содержит ряд недостатков, не позволяющих мне засчитать работу на данной итерации:

Регулярные выражения:
Имя:
**Нужно исправить:** Валидацию проходят следующие некорректные кейсы:
Денис- (дефис, нет двойного имени, допустимо либо Денис, либо Денис-Иван)
Денис-ловеркейс (второе имя должно быть с заглавной буквы)

Email:
**Нужно исправить:** Валидацию не проходят следующие корректные кейсы:
t@gmail.com (однобуквенное имя пользователя является корректным)

Адрес сайта:
**Нужно исправить:** валидацию проходит следующий некорректный адрес сайта:
http://ya.ru:00000 (некорректный номер порта)

**Comment by zlocate** [comment]
Нужно исправить:
Не проходят валидацию корректные URL с номером порта
http://ya.ru:8000
http://8.8.8.8:8000

**Comment by romanRysev** [comment]
Все это уже было исправлено. Сейчас тоже работает.

**Comment by zlocate** [comment]
Добрый день. В текущей ревизии проблема решена, отлично.

**Можно лучше**: Расходится дерево коммитов между `master` и `gh-pages`. (разные версии страницы, изменения из `master` не попали) в репозиторий.
Просьба сливать изменения в `gh-pages` после внесения правок в других ветках.

Небольшой комментарий:
Ревьювер проверяет версию, расположенную на Github Pages (ветка `gh-pages`).
Данные проблемы были исправлены только в последнем коммите в ветке `gh-pages`.
На момент проверки ревьювер анализировал версию
```
commit 7e3e313cc5df8e7d10fc78af5641fb11bdc2f948 (HEAD)
Author: Roman Rysev <roman.rysev@gmail.com>
Date:   Sun Mar 15 00:24:47 2020 +0300

    create main files
``` 

В любом случае у нас нет цели найти у вас ошибки, которых нет.
Мы искренне хотим чтобы у вас все получилось и вы вынесли максимум из учебного процесса, однако также работаем в рамках требований, предъявляемых условием задания.

К сожалению, одна из проблем отмеченная ранее не исправлена в текущей версии :(
**Нужно исправить:** Текущая версия из ветки master и версия, размещенная на Github Pages пропускает следующий некорректный адрес:
http://ya.ru:00000 (некорректный номер порта)

Скриншоты:
https://prnt.sc/rh7gdl (версия на Github Pages)
https://prnt.sc/rh7his  (версия из ветки `master`)

После исправления работа будет засчитана.

**Comment by romanRysev** [comment]
Добрый день.
Да, прозевал. Сейчас исправил в обеих ветках.

**Comment by zlocate** [comment]
К сожалению, теперь не проходит валидацию URL c корректным портом, например
`http://ya.ru:22100`
Допустимые значения 1-65536. 
Однако достаточно чтобы не проходили нулевые значения (первая цифра не ноль) и проходили любые числа состояцие из 1-5 цифр.

**Comment by zlocate** [comment]
Спокойно дорабатывайте этот момент, данную итерацию проверки я не засчитываю

**Comment by romanRysev** [comment]
Исправил)

**Comment by zlocate** [comment]
Добрый день. 
**Нужно исправить:**
Кейс выше починился, но остался еще один:
Обратимся к заданию:
`Порт начинается с двоеточия, за которым идут от 2 (!) до 5 цифр. Например: :8080;`
Сейчас валидацию проходит порт даже из 1й цифры :(
https://prnt.sc/rhn4tl
Как только поправите, я зачту.

**Comment by romanRysev** [comment]
Странно, судя по Вики и Вашему прошлому комменту - порты могут быть и из 1 цифры (в т.ч. 0)

**Comment by romanRysev** [comment]
Исправил.

**Comment by zlocate** [comment]
Да, прошу прощения за неточность. Фактически, разумеется, порт может быть и одноциферный, к сожалению в задании ставится такое ограничение. Я отправил запрос на исправление текста задания. 
В следующей ревизии задания это будет исправлено.
Надеюсь этот недочет не испортит вам впечатление от учебного процесса :)

Задание зачтено. Желаю вам успехов в дальнейшем прохождении курса!
Денис, команда Яндекс.Практикум.

**Comment by romanRysev** [comment]
Без проблем) Спасибо!
